src/utils/metrics.cpp

```diff
@@ -449,6 +444,89 @@ metric_entity_ptr metric_registry::find_or_create_entity(const metric_entity_pro
     return entity;
 }
 
+DSN_DECLARE_string(cluster_name);
+
+namespace {
+
+#define ENCODE_UNKNOWN_IF(expr)                                                                    \
+    do {                                                                                           \
+        if (dsn_unlikely(expr)) {                                                                  \
+            dsn::json::json_encode(writer, "unknown");                                             \
+            return;                                                                                \
+        }                                                                                          \
+    } while (0)
+
+void encode_cluster(dsn::metric_json_writer &writer)
+{
+    writer.Key(dsn::kMetricClusterField.c_str());
+
+    ENCODE_UNKNOWN_IF(utils::is_empty(dsn::FLAGS_cluster_name));
+
+    dsn::json::json_encode(writer, dsn::FLAGS_cluster_name);
```

Comment:
How about encapsulate the 2 lines to a single line, the judgment can be put into the macro.

The other places are the same.

Reply:
OK.